clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -487,7 +487,12 @@ else if (strategy == OffsetResetStrategy.LATEST)
                 } else {
                     TopicPartition partition = nextInLineRecords.partition;
                     fetchedPartition = partition;
-                    fetchedOffsets = subscriptions.position(partition);
+                    if (subscriptions.isAssigned(partition)) {
```

Comment:
I think we need to skip all the code in this else block if the partition is no longer assigned.